NoBeardProject/src/main/java/machine/NoBeardMachine.java

```diff
@@ -89,9 +94,37 @@ public void loadProgram(int atAddr, byte[] data) {
     public void runProgram(int startPc) {
         System.out.println("Starting programm at pc " + startPc);
         controlUnit.startMachine(startPc);
-        while (getState() == ControlUnit.MachineState.RUNNING) {
-            step();
+        breakpointIndex = 0;
+        jumpToNextBreakpoint();
+    }
+
+    public void jumpToNextBreakpoint() {
+        if (breakpointIndex < breakpoints.size()) {
+            while (controlUnit.getPc() < breakpoints.get(breakpointIndex)) {
```

Comment:
This will cause problems if there is a halt instruction before a breakpoint. This could happen, e.g., if a programmer does a plausibility check on input data and if this fails (s)he wants to stop the program.